File: programmers_문자열압축.py
def solution(s):
    answer = ''
    for i in range(1,len(s)//2+1):
        now_ans = ''
        j=0
        while j<len(s):
            cnt=1
            X=s[j:j+i]
            while s[j:j+i] == s[j+i:j+2*i]:
                cnt+=1
                j=j+i
            if cnt>1:
                now_ans += str(cnt)
            now_ans += X
            j+=i
        if answer == '' or len(answer) > len(now_ans):
            answer = now_ans
    if answer == '':
        answer = s
    return len(answer)

print(solution('aabbaccc'))
print(solution("ababcdcdababcdcd"))

File: programmers_문자열_압축.js
function solution(s) {
  var answer = "";
  for (var i = 1; i < s.length / 2 + 1; i++) {
    let innerAnswer = "";
    let j = 0;
    console.log(s.substr(j, j + i));
    while (j < s.length) {
      let cnt = 1;
      let X = s.substr(j, i);
      while (s.substr(j, i) === s.substr(j + i, i)) {
        cnt = cnt + 1;
        j = j + i;
      }
      if (cnt > 1) {
        innerAnswer = innerAnswer + cnt.toString();
      }
      innerAnswer = innerAnswer + X;
      j = j + i;
    }
    if (answer === "" || answer.length > innerAnswer.length) {
      answer = innerAnswer;
    }
  }
  if (answer === "") answer = s;
  return answer.length;
}
